yarn-project/sequencer-client/src/sequencer/public_processor.ts

```diff
@@ -91,6 +90,20 @@ export class PublicProcessor {
     const publicKernelOutput = await this.publicKernel.publicKernelCircuitNoInput(publicKernelInput);
     const publicKernelProof = await this.publicProver.getPublicKernelCircuitProof(publicKernelOutput);
 
+    const contractTreeInfo = await this.db.getTreeInfo(MerkleTreeId.CONTRACT_TREE);
+    const privateDataTreeInfo = await this.db.getTreeInfo(MerkleTreeId.PRIVATE_DATA_TREE);
+    const nullifierTreeInfo = await this.db.getTreeInfo(MerkleTreeId.NULLIFIER_TREE);
+
+    publicKernelOutput.constants.historicTreeRoots.privateHistoricTreeRoots.nullifierTreeRoot = Fr.fromBuffer(
```

Comment:
@spalladino we had to add these as there is a validation step in the circuit block builder that failed otherwise. It looks like there is a bit of a hack in aztec node to supply these values for private functions.

Reply:
So I think we should not be adding these manually to the kernel circuit output, but rather we should make sure they are available as part of the inputs, and the circuit should be forwarding them to the outputs.

I think the problem is that the `PublicKernelInputsNoPreviousKernel`, having no previous kernel, has no access to the `constants`, so it never receives the tree roots as input and cannot forward them to the outputs (as the private kernel circuit does [here](https://github.com/AztecProtocol/aztec3-packages/blob/151b2169277c2a43c811f0fa268b36cb51ad6e07/circuits/cpp/src/aztec3/circuits/kernel/private/native_private_kernel_circuit.cpp#L58)).

The fix would be to modify the `PublicKernelInputsNoPreviousKernel` so it also receives a `CombinedConstantData` which is used to populate the outputs. However, if we're removing this circuit, maybe we can just live with this hack a bit longer.

WDYT @PhilWindle?

Reply:
I think the plan is to modify the private kernel circuit so it also has a 'no previous kernel' mode which would result in the same issue. A question for @iAmMichaelConnor really. In the case of 'no previous kernel' iterations, should the constant data (tree roots) be supplied as an input to the circuit so the circuit can forward it to the circuit's public inputs? Currently the caller of the circuit (aztec rpc or sequencer) has to enrich this data after the 'no previous kernel' iteration.